### PR TITLE
HPCC-25182 handle persistent volume path with escape  char "\"

### PR DESCRIPTION
**Pull request by xwang2713**
To deploy HPCC cluster with persistent volume using startall.sh on Windows:
   ./startall.sh -l \<image tag\> -p \<hostpath\>
For Docker Desktop (DD) with Hypber-V <hostpath> probably should be "\\/<drive>\\/<data directory>". For example "\\/c\\/hpcc-data". Otherwise passing "-p /c/\<data directory\>" to startall.sh will result hostpath C:/<data directory> in Persistent Volume (PV) in Kubernetes which will not be recognized by pod.
But mkdir command in startall.sh can't handle path "\\/<drive>\\/<data directory>" and "\\" char should be removed.

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

@Michael-Gardner please review

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested on local environment DockerDesktop environment with git bash.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 looks good

**Comment by richardkchapman** [comment]
Missing lots of checkboxes. In particular, look at the one that says "The commit message title makes sense in a changelog, by itself" and ask whether it's true. I'd say without looking at the actual change the commit title makes no sense at all...

File: dockerfiles/startall.sh [hunk 1]
```diff
@@ -60,9 +60,10 @@ done
 [[ -z ${LABEL} ]] && LABEL=$(docker image ls | fgrep "${DOCKER_REPO}/platform-core" | head -n 1 | awk '{print $2}')
 
 if [[ -n ${PERSIST} ]] ; then
-  mkdir -p ${PERSIST}/dlls
-  mkdir -p ${PERSIST}/dali
-  mkdir -p ${PERSIST}/data
+  PERSIST_PATH=$(echo $PERSIST | sed 's/\\//g')
```
Comment (Michael-Gardner) [inline dockerfiles/startall.sh:63]: replacement works

**Review by Michael-Gardner — APPROVED** [review]
@xwang2713 looks good.  Richard commented that the commit message should be changed. Perhaps 'Fixed backslash characters in startall.sh scripts PERSIST variable causing issues in bash'?

**Review by Michael-Gardner — APPROVED** [review]
@xwang2713 my approval got wiped from the checklist when you changed the commit message. Re-approving and bumping the PR.

-- Actually I think that message means Richard or Gavin has to approve of it. Says 'required by reviewers with write access'.

**Comment by richardkchapman** [comment]
@xw this is fine to merge, but the commit message needs addressing so that the changelog looks sensible. When deciding on a commit message title, imagine someone looking at the changelog and trying to decide whether a particular change might affect them...

**Comment by xwang2713** [comment]
@richardkchapman I update commit message as well as PR title and message.
I try to rebase but still see "Some checks haven’t completed yet". I am not sure if I mess up something. If it is let me know I can re-create the PR.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**